### PR TITLE
feat(typescript): add typescript support to the repo

### PR DESCRIPTION
**Pull request by PeterMK85**
### Description

_Describe the changes and motivations for the pull request, unless obvious from the title._

### Review

- [x] Verify that all classes affected by the changes have **good** top-level documentation.
- [ ] Verify that new code has been covered with specs (and features if applicable).
- [ ] Verify that a corresponding meaningful changelog entry has been added (`* Add Github pull request template`), for this change if applicable.
- [ ] Test manually.
- [ ] Get two :+1: from code review.

### Pre-merge checklist

- [ ] The PR relates to a single subject with a clear title and description in grammatically correct, complete sentences.
- [ ] Verify that all new dependencies are included in `package.json`.
- [ ] Double check the quality of [commit messages](http://chris.beams.io/posts/git-commit/).
- [ ] Squash related commits together.
- [ ] Verify the branch name starts with `NNNN` where `NNNN` is the issue number.

### Screenshots

| Before                                        | After                                         |
| --------------------------------------------- | --------------------------------------------- |
| _Insert screenshots and/or screen recordings_ | _Insert screenshots and/or screen recordings_ |

### Other

_Provide additional notes, remarks, links, mention specific people to review,…_

### CODE REVIEW

File: src/calendar/day.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    isWeekend,
+    isWorkday,
+  } = props
+  const children = formatDate(date, 'D')
+
+  if (customRender) {
+    return customRender({
+      ...props,
+      // TODO: remove this and leave it to the passed function to format the date in the desired format
+      children,
```
Comment (PeterMK85) [inline src/calendar/day.tsx]: @havenchyk 
I think in the future, we should only pass non formatted date to the render prop, and the render prop itself will have the full freedom to format prior to implemention

File: src/calendar/day.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import cx from 'classnames'
```
Comment (PeterMK85) [inline src/calendar/day.tsx]: 1-1 transform from the previous file `src/calendar/day.jsx` no logical change has been applied

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "compilerOptions": {
+    "allowJs": true,
+    "allowSyntheticDefaultImports": true,
+    "baseUrl": "./src",
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "isolatedModules": true,
```
Comment (havenchyk) [inline tsconfig.json]: I didn't know about this option, but it looks like for using with babel typescript plugin it's a must.

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    "resolveJsonModule": true,
+    "skipLibCheck": true,
+    "strict": true,
+    "target": "es5",
+    "typeRoots": ["node_modules/@types", "src/@types"],
+    "paths": {
+      "*": ["@types/*"]
+    },
+  },
+  "include": ["src/**/*"],
```
Comment (havenchyk) [inline tsconfig.json]: can be just `src` iirc

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    "isolatedModules": true,
+    "jsx": "preserve",
+    "lib": ["dom", "dom.iterable", "esnext"],
+    "module": "esnext",
+    "moduleResolution": "node",
+    "noEmit": true,
+    "resolveJsonModule": true,
+    "skipLibCheck": true,
+    "strict": true,
+    "target": "es5",
```
Comment (havenchyk) [inline tsconfig.json]: we use babel, so target can be `es2017`

File: tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "isolatedModules": true,
+    "jsx": "preserve",
+    "lib": ["dom", "dom.iterable", "esnext"],
+    "module": "esnext",
+    "moduleResolution": "node",
+    "noEmit": true,
+    "resolveJsonModule": true,
+    "skipLibCheck": true,
```
Comment (havenchyk) [inline tsconfig.json]: I'd remove this prop

File: tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  "compilerOptions": {
+    "allowJs": true,
+    "allowSyntheticDefaultImports": true,
+    "baseUrl": "./src",
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "isolatedModules": true,
+    "jsx": "preserve",
+    "lib": ["dom", "dom.iterable", "esnext"],
+    "module": "esnext",
```
Comment (havenchyk) [inline tsconfig.json]: module most likely needs to be `node`

File: src/calendar/day.tsx (hunk 1 above)
Comment (havenchyk) [reply src/calendar/day.tsx]: it sounds good, initially I wanted to keep backward compatibility

File: .babelrc.js [hunk 8]
```diff
@@ -1,4 +1,14 @@
...
-  presets: ['@babel/preset-env', '@babel/preset-react'],
-  plugins: ['@babel/plugin-proposal-class-properties'],
+  presets: [
+    '@babel/preset-env',
+    ['@babel/preset-react',
+      {
+        development: process.env === 'development' || process.env === 'test',
+      }
+    ],
+    '@babel/typescript'
```
Comment (havenchyk) [inline .babelrc.js]: `@babel/preset-typescript`

File: tsconfig.json (hunk 7 above)
Comment (PeterMK85) [reply tsconfig.json]: @havenchyk 

`Argument for '--module' option must be: 'none', 'commonjs', 'amd', 'system', 'umd', 'es6', 'es2015', 'esnext'.`

**Pull request merged**